servers/navigation_common.h

```diff
@@ -1,5 +1,5 @@
 /**************************************************************************/
-/*  nav_heap.h                                                            */
+/*  navigation_common.h                                                   */
```

Comment:
That sole navigation file under `servers/` seems a bit weird.

We should consider if this alternative isn't better:
- Duplicate the shared enums in both the 2D and 3D folders, it's just a few constants and not uncommon in the Godot codebase to do this.
- Move the Heap class to `core`.

Reply:
CC @smix8

Reply:
Another option is I could just move it into, say, the NavigationServer3D folder, and have this header be the only dependency 2D navigation has on 3D. Since it's not a cpp file, the build system wouldn't have to know. 🙈

Reply:
For now I'd say keep it as `server/nav_heap.h` and move the enums back, duplicating them in 2D and 3D (maybe re-creating `navigation_utilities_{2,3}d.h` out of `navigation_defaults_{2,3}d.h` + these enums).

Reply:
I got sidetracked with scope creep so I opened PR #105718 which we can merge first.

Reply:
Done.

Reply:
Another option is we could have `servers/navigation/2d/` and `servers/navigation/3d/` instead of `servers/navigation_2d/` and `servers/navigation_3d/`, putting the common files in `servers/navigation/`. But then there is the question of if we would do this with physics as well.